packages/cli/src/commands/generate.ts

```diff
@@ -41,33 +49,53 @@ export class GenerateCommand extends Command {
   });
 
   reportName = Option.String("--report-name,--name", {
-    description: "The report name",
+    description: "The report name (default: Allure Report)",
+  });
+
+  stage = Option.Array("--stage", {
+    description: "Stages archives to restore state from (default: empty string)",
   });
 
   async execute() {
-    const cwd = this.cwd ?? process.cwd();
+    const cwd = this.cwd ?? processCwd();
     const resultsDir = (this.resultsDir ?? "./**/allure-results").replace(/[\\/]$/, "");
     const config = await readConfig(cwd, this.config, {
       name: this.reportName,
       output: this.output ?? "allure-report",
     });
-    const matcher = pm(resultsDir, {
-      dot: true,
-      contains: true,
-    });
-    const resultsDirectories = new Set<string>();
-
-    await findMatching(cwd, resultsDirectories, (dirent) => {
-      if (dirent.isDirectory()) {
-        const fullPath = join(dirent?.parentPath ?? dirent?.path, dirent.name);
-
-        return matcher(fullPath);
+    const stageDumpFiles: string[] = [];
+    const resultsDirectories: string[] = [];
+
+    if (this.stage?.length) {
+      for (const stage of this.stage) {
+        const matchedFiles = await glob(stage, {
+          nodir: true,
+          dot: true,
+          absolute: true,
+          cwd,
+        });
+
+        stageDumpFiles.push(...matchedFiles);
       }
+    }
 
-      return false;
-    });
+    // don't read allure results directories without the parameter when stage file has been found
+    // or read allure results directory when it is explicitly provided
+    if (!!this.resultsDir || stageDumpFiles.length === 0) {
+      const matchedDirs = (
+        await glob(resultsDir, {
```

Comment:
Let's also use `windowsPathsNoEscape` (at least on Windows). Otherwise, it's too surprising to get `No test results directories found matching pattern...` on Windows e.g., when using tab autocompletion (which replaces all dir separators to `\`).